plugins/modules/user_role_workflow_manager.py

```diff
@@ -3265,31 +3322,58 @@ def main():
     ccc_user_role = UserandRole(module)
     state = ccc_user_role.params.get("state")
 
-    if "role_details" in ccc_user_role.params.get("config"):
-        ccc_user_role.validate_input_yml(ccc_user_role.params.get("config").get("role_details")).check_return_status()
-        config_verify = ccc_user_role.params.get("config_verify")
-
-        for config in ccc_user_role.validated_config:
-            ccc_user_role.reset_values()
-            ccc_user_role.get_want(config).check_return_status()
-            ccc_user_role.get_have(config).check_return_status()
-            ccc_user_role.get_diff_state_apply[state](config).check_return_status()
-
-            if config_verify:
-                ccc_user_role.verify_diff_state_apply[state](config).check_return_status()
-
-    if "user_details" in ccc_user_role.params.get("config"):
-        ccc_user_role.validate_input_yml(ccc_user_role.params.get("config").get("user_details")).check_return_status()
-        config_verify = ccc_user_role.params.get("config_verify")
-
-        for config in ccc_user_role.validated_config:
-            ccc_user_role.reset_values()
-            ccc_user_role.get_want(config).check_return_status()
-            ccc_user_role.get_have(config).check_return_status()
-            ccc_user_role.get_diff_state_apply[state](config).check_return_status()
-
-            if config_verify:
-                ccc_user_role.verify_diff_state_apply[state](config).check_return_status()
+    if state == "merged":
```

Comment:
Code is repeated in multiple places.. Can you optimize it? Can you write an API? 

Please change the name of the API appropriately. please kindly check the logic .. I pasted sample code..
 
```
if state == "merged":
    process_config_details("role_details")
    process_config_details("user_details")
else:
    process_config_details("user_details")
    process_config_details("role_details")

def process_config_details(config_type):
    if config_type in ccc_user_role.params.get("config"):
        ccc_user_role.validate_input_yml(ccc_user_role.params.get("config").get(config_type)).check_return_status()
        config_verify = ccc_user_role.params.get("config_verify")

        for config in ccc_user_role.validated_config:
            ccc_user_role.reset_values()
            ccc_user_role.get_want(config).check_return_status()
            ccc_user_role.get_have(config).check_return_status()
            ccc_user_role.get_diff_state_apply[state](config).check_return_status()

            if config_verify:
                ccc_user_role.verify_diff_state_apply[state](config).check_return_status()

```

Reply:
Updated the code accordingly.